crates/uv/src/commands/tool/install.rs

```diff
@@ -362,13 +382,30 @@ pub(crate) async fn install(
                         environment.interpreter().sys_executable().display()
                     );
                     true
-                } else {
+                } else if explicit_python_request {
                     let _ = writeln!(
                         printer.stderr(),
                         "Ignoring existing environment for `{from}`: the requested Python interpreter does not match the environment interpreter",
                         from = from.name.cyan(),
                     );
                     false
+                } else {
+                    // Allow the existing environment if the user didn't explicitly request another
+                    // version
+                    if let Some(ref tool_receipt) = existing_tool_receipt {
+                        if settings.reinstall.is_all() && tool_receipt.python().is_none() && python_request.is_some() {
+                            let _ = writeln!(
+                                printer.stderr(),
+                                "Ignoring existing environment for `{from}`: the Python interpreter does not match the environment interpreter",
+                                from = from.name.cyan(),
+                            );
+                            false
+                        } else {
+                            true
+                        }
+                    } else {
+                        true
+                    }
```

Comment:
I think this could be refactored to avoid the repeated `else` branches, but the logic should be correct. The comment needs more detail.